linuxbrew-common.sh

```diff
@@ -0,0 +1,38 @@
+#@IgnoreInspection BashAddShebang
+
+# Copyright (c) YugaByte, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied.  See the License for the specific language governing permissions and limitations
+# under the License.
+#
+
+set -euo pipefail
+
+if [[ $BASH_SOURCE == $0 ]]; then
+  echo "$BASH_SOURCE must be sourced, not executed" >&2
+  exit 1
+fi
+
+declare -i -r ABS_PATH_LIMIT=85
+
+get_brew_link() {
+  local brew_dirname="linuxbrew-$(date +%Y%m%dT%H%M%S)"
+  local brew_link="$(realpath .)/$brew_dirname"
+  local len=${#brew_link}
+  [[ $len -le $ABS_PATH_LIMIT ]] || (echo "Linuxbrew link absolute path should be no more than\
+ $ABS_PATH_LIMIT bytes, but actual length is $len bytes: $brew_link"; exit 1)
```

Comment:
I would write these conditions like
```bash
if [[ $len -gt $ABS_PATH_LIMIT ]]; then
  echo "Linuxbrew link absolute path should be no more than $ABS_PATH_LIMIT bytes," \
       "but actual length is $len bytes: $brew_link" >&2
  exit 1
fi
```
Then we don't have to think about "not less than equal" -> "actually this errors out when the length is greater". Also error output should go to stderr.
Also I would like to point out that the original code above seems to rely on `set -e` being there, because `exit 1` would only exit from the subshell, and the outer command will return a non-zero exit code, which due to `set -e` will result in the script exiting. A more straightforward `if` statement works independently of `set -e`.

Also note that `echo` will concatenate multiple string arguments given to it and put a space in between, so it is perfectly reasonable to split long `echo` arguments as specified in my example above.